Modules/Core/Common/include/itkOptimizerParameters.hxx

```diff
@@ -23,70 +23,51 @@
 
 namespace itk
 {
-/** Default contstructor */
+/** Default constructor */
 template <typename TParametersValueType>
-OptimizerParameters<TParametersValueType>::OptimizerParameters()
-  : Array<TParametersValueType>()
-  , m_Helper(nullptr)
-{
-  this->Initialize();
-}
+OptimizerParameters<TParametersValueType>::OptimizerParameters() = default;
 
 /** Copy constructor */
 template <typename TParametersValueType>
 OptimizerParameters<TParametersValueType>::OptimizerParameters(const OptimizerParameters & rhs)
   : Array<TParametersValueType>(rhs)
-  , m_Helper(nullptr)
 {
   // Note: don't copy the OptimizerParametersHelper.
   // The Array copy constructor will allocate new memory
   // and copy the data to it. So we end up here with a generic
   // OptimizerParameters data object even if 'rhs' points to
   // something different.
-  this->Initialize();
 }
 
 /** Constructor with size */
 template <typename TParametersValueType>
 OptimizerParameters<TParametersValueType>::OptimizerParameters(SizeValueType dimension)
   : Array<TParametersValueType>(dimension)
-  , m_Helper(nullptr)
-{
-  this->Initialize();
-}
+{}
 
-/** Constructor with Array assignment */
+/** Constructor with Array */
 template <typename TParametersValueType>
 OptimizerParameters<TParametersValueType>::OptimizerParameters(const ArrayType & array)
   : Array<TParametersValueType>(array)
-  , m_Helper(nullptr)
-{
-  this->Initialize();
-}
+{}
 
 template <typename TParametersValueType>
 void
 OptimizerParameters<TParametersValueType>::Initialize()
 {
   // Set the default OptimizerParametersHelper
-  auto * helper = new OptimizerParametersHelperType;
-  // OptimizerParameters will manage this memory.
-  this->SetHelper(helper);
+  this->m_Helper.reset(new OptimizerParametersHelperType);
 }
 
 /** Destructor */
 template <typename TParametersValueType>
-OptimizerParameters<TParametersValueType>::~OptimizerParameters()
-{
-  delete this->m_Helper;
-}
+OptimizerParameters<TParametersValueType>::~OptimizerParameters() = default;
 
 template <typename TParametersValueType>
 void
 OptimizerParameters<TParametersValueType>::SetHelper(OptimizerParametersHelperType * helper)
 {
-  delete this->m_Helper;
-  this->m_Helper = helper;
+  this->m_Helper.reset(helper);
```

Comment:
As all of the methods are now one-liners, maybe we no longer need .hxx file and could move all the implementations into .h file?

Reply:
If would be good if others pitched in with opinions.

Reply:
@dzenanz  If some people still use `ITK_MANUAL_INSTANTIATION`, it may matter. Compilation, link, or run time might then be affected by moving code between "h" and "hxx". (Not sure in what way the duration would be affected!) Do people still use  `ITK_MANUAL_INSTANTIATION`?

Another effect might be that the compiler might derive automatically that a constructor or a destructor is `noexcept` or `constexpr`, when it is defaulted ( `= default`) in the "h file", instead of in the hxx file.

Reply:
Correction: a destructor is always `noexcept` by default. But it might be easier for the compiler to figure out that the destructor won't terminate the program (in case of an exception within the destructor), when the destructor is defaulted in the "h file", rather than in the hxx file.

Reply:
Personally I would like to have the entire implementation of `OptimizerParameters` in the .h file, and inside the class definition. But of course, it depends on what others think... 😃

Reply:
@N-Dekker I agree that the trivial nature of the content in the .hxx file does not justify the overhead.  In general, I do like large, non-trivial functions to be placed in the .hxx file for readability purposes.